### ISSUE TITLE
[BUG] Trace files made with Firefox contain broken UTF characters

### ISSUE THREAD

**Issue by kachkaev** [microsoft/playwright#16601]
This bug is likely related to https://github.com/microsoft/playwright/issues/16367. I decided create a separate issue as it occurs in a different scenario and may be more impactful.

**Context:**
- Playwright Version: 1.25.0
- Operating System: macOS
- Node.js version: 16.16.0
- Browser: Firefox

**Code Snippet**

```javascript
import { chromium, firefox, webkit } from "playwright";

for (const engine of [chromium, firefox, webkit]) {
  console.log(engine.name());

  const browser = await engine.launch();
  const context = await browser.newContext();
  await context.tracing.start({ screenshots: true, snapshots: true });

  const page = await context.newPage();

  await page.goto(
    "https://ru.wikipedia.org/wiki/%D0%94%D1%80%D0%B0%D0%BC%D0%B0%D1%82%D1%83%D1%80%D0%B3%D0%B8%D1%8F",
  );

  const traceFilePath = `trace-goto-${engine.name()}.zip`;
  await context.tracing.stop({ path: traceFilePath });

  await browser.close();
  console.log(`→ ${traceFilePath}`);
}
```

**Describe the bug**

Recording a trace file with Firefox creates broken UTF characters in the snapshots (`�`). This means that certain Cyrillic, Greek, Chinese etc. words end up broken. For example, ` народная` becomes ` нар��дная` in a [random Wikipedia article I’ve picked](https://ru.wikipedia.org/wiki/%D0%94%D1%80%D0%B0%D0%BC%D0%B0%D1%82%D1%83%D1%80%D0%B3%D0%B8%D1%8F).

✅ [trace-goto-chromium.zip](https://github.com/microsoft/playwright/files/9355359/trace-goto-chromium.zip)  
✅ [trace-goto-webkit.zip](https://github.com/microsoft/playwright/files/9355360/trace-goto-webkit.zip)  

<img width="1348" alt="Screenshot 2022-08-17 at 01 01 22" src="https://user-images.githubusercontent.com/608862/185006130-151e8a4c-6b99-41c9-afbb-e95f73236ffd.png">

❌ [trace-goto-firefox.zip](https://github.com/microsoft/playwright/files/9355371/trace-goto-firefox.zip)
<img width="1348" alt="Screenshot 2022-08-17 at 01 01 09" src="https://user-images.githubusercontent.com/608862/185006146-3c32e537-e097-4d5c-bb4c-0b22fdf2dedd.png">

Another way to reproduce this issue is by replacing `page.goto` with `page.evaluate`:

```javascript
import { chromium, firefox, webkit } from "playwright";

for (const engine of [chromium, firefox, webkit]) {
  console.log(engine.name());

  const browser = await engine.launch();
  const context = await browser.newContext();
  await context.tracing.start({ screenshots: true, snapshots: true });

  const page = await context.newPage();

  await page.evaluate(() => {
    document.body.innerHTML = "🎭".repeat(10_000);
  });

  const traceFilePath = `trace-evaluate-${engine.name()}.zip`;
  await context.tracing.stop({ path: traceFilePath });

  await browser.close();
  console.log(`→ ${traceFilePath}`);
}
```

✅ [trace-evaluate-chromium.zip](https://github.com/microsoft/playwright/files/9355385/trace-evaluate-chromium.zip)
✅ [trace-evaluate-webkit.zip](https://github.com/microsoft/playwright/files/9355386/trace-evaluate-webkit.zip)

<img width="1348" alt="Screenshot 2022-08-17 at 00 48 09" src="https://user-images.githubusercontent.com/608862/185006831-32fdf151-9c83-4496-a49e-0bbe65358110.png">

❌ [trace-evaluate-firefox.zip](https://github.com/microsoft/playwright/files/9355387/trace-evaluate-firefox.zip)
<img width="1348" alt="Screenshot 2022-08-17 at 00 48 24" src="https://user-images.githubusercontent.com/608862/185006850-33ae1e30-ba7b-4211-8aaf-841f7d500d01.png">
<div align=center><em>more cases are seen if you scroll the snapshot</em></div>

<br/><br/>
Fixing this bug will likely close https://github.com/microsoft/playwright/issues/16367 too.

**Comment by aslushnikov** [comment]
Let me merge this into #16367; I'll take a look on that one as well.

**Comment by kachkaev** [comment]
👋 @aslushnikov! Merging two issues makes sense – I’ve opened this one thinking that it'd supersede #16367 based on impact, but either way is fine.

Is it practical to fix this is a 1.25 patch or is the fix too involved?

I'm using Playwright in a time-sensitive [pro-bono project](https://github.com/kachkaev/special-internet-archive) and am somewhat blocked by this bug. Playwright is used to capture pretty large web pages as trace files and Firefox seems to be the only choice. Chromium crashes after a certain page size (VK feed length) and Webkit cannot be opened in a ‘no-image’ mode to keep trace files small.

If there is a way to increase Chromium’s memory cap or load Webkit with something like `args: ["--blink-settings=imagesEnabled=false"]` / `firefoxUserPrefs: { "permissions.default.image": 2 }` in FF, that might help too.
